pkg/kv/kvserver/replica_app_batch.go

```diff
@@ -415,13 +415,9 @@ func (b *replicaAppBatch) runPostAddTriggersReplicaOnly(
 		// going to bother with a long-running migration.
 		apply := !looselyCoupledTruncation || res.RaftExpectedFirstIndex == 0
 		if apply {
-			// TODO(pavelkalinnikov): counting bytes may be unreliable because 0 bytes
-			// does not mean 0 files. Make the interface more robust.
-			if rmBytes, _, err := b.r.raftMu.sideloaded.BytesIfTruncatedFromTo(
-				ctx, 0, res.State.TruncatedState.Index,
-			); err != nil {
-				return errors.Wrap(err, "unable to count sideloaded files to remove")
-			} else if rmBytes != 0 {
+			if has, err := b.r.raftMu.sideloaded.HasEntries(ctx, 0, res.State.TruncatedState.Index); err != nil {
```

Comment:
Might need to move this check to `handleTruncatedStateBelowRaftPreApply`, and check `HasEntries` in `[b.state.TruncatedState, res.State.TruncatedState.Index)` instead of `[0, res.State.TruncatedState.Index)`.

Counter-intuitively though, the check from 0 is more efficient because it may terminate more early. It may result in a false-positive sync though.

Reply:
Not moving into `handleTruncatedStateBelowRaftPreApply` because it is used by the loosely coupled truncations path, which already ensures syncs and doesn't need this extra check.

Moved `HasEntries` to be after `handleTruncatedStateBelowRaftPreApply` though, because the latter can withdraw the truncation command.